### DISCUSSION TITLE
Trainer.fit validating before finishing current training epoch

### DISCUSSION THREAD

**Discussion by minwang-ai** [Lightning-AI/pytorch-lightning#10586] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
When the train.fit() starts to train my model, it starts validating at 95% of training of current epoch instead of waiting until 100%.

Why does it happen? It makes Val-loss/acc/miou not accurate anymore…
![79A18250-AF4C-4F48-BB41-EB930C6ACA7F](https://user-images.githubusercontent.com/75472853/142167568-d02d1ae6-5886-4d9e-bd06-7bc37e3a3e28.jpeg)

**Comment by rohitgr7** [comment, marked as answer]
the master progress bar that shows you the progress of the corresponding epoch consists of both train and val steps. So it completes the training within the first 95% of the total steps and the last 5% is completed for validation.
